### DISCUSSION TITLE
Calling another .py from the converted .exe file

### DISCUSSION THREAD

**Discussion by ntahlahweh** [TomSchimansky/CustomTkinter#2579] [Q&A] [answered]
Hi everyone, I just want to ask how can I call another `.py` from the converted `.exe` file?

I have a main python code called `main_2.py` file that act as the main interface for program that I created which in this case is the one that will be converted into `.exe`. One of its main function is to open up another python code called `analysis.py` which in this case is another GUI itself that will do other task.  The order of the file could be seen from the directory here:

![Screenshot 2024-09-16 130943](https://github.com/user-attachments/assets/076b7bc7-79e4-4e0b-9df0-c98a389a022c)

My question is can the converted `.exe` file called the `.py` directly or should I convert the `analysis.py` into `.exe` as well?

Thank you

**Comment by Akascape** [comment]
@ntahlahweh 
Solution1: pack the other py files as a module and import in main application, then link them with the main app.
Solution2: Convert the other py file to separate exe, call it through subprocess. Use if both app are not linked together.

You can't directly run the other py file through the converted exe if the system doesn't have python compiler installed. So there will be no point of executable if that's the case.

**Reply by ntahlahweh** [reply]
Hi! Just want to update, I discovered this [How do I Convert All My Python Files?](https://nitratine.net/blog/post/issues-when-using-auto-py-to-exe/#how-do-i-convert-all-my-python-files) and according to this the tool will look for import statements and include those files and libraries too (where possible) and I did it by specifically making sure the location of the `.py` code is properly call in order for it to works!

Anyway the solution 2 provided also work as intended either way.

**Comment by ntahlahweh** [comment]
I think Solution 2 might be the best option for me. Just out of curiosity, how does solution 1 work? Is there any example that I can see?

Another thing, I realised when I converted from `.py` to .`exe` there is an "_internal" folder created. So if I proceed with the solution 2 there will be multiple "_internal" folder. My question is, can I combine the "_internal" folder together in one place? 

Thank you

**Reply by Akascape** [reply, marked as answer]
> My question is, can I combine the "_internal" folder together in one place?

@ntahlahweh 
I haven't tried this before, but it should work since both apps use the same internal files packaged by PyInstaller. Compare the contents of both internal folders to ensure compatibility. If the second executable doesn't work, try copying any missing files from its internal folder. 

Or you can also try one-file mode for the second exe, check this post for more info: #939

**Comment by ntahlahweh** [comment]
I tried the solution 2 and it works! I also did place combine the `_internal` folder in one place and no issue so far. The converted `.exe` file work as intended. I will test further and report here.

As for the one-file mode, it is interesting idea. I will try to convert it and see how it goes
